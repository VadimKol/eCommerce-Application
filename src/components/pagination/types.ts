export interface PaginationProps {
  page: number;
  setPage: (page: number) => void;
  setLoadingProducts: (loadingProducts: boolean) => void;
  total: React.MutableRefObject<number>;
}
