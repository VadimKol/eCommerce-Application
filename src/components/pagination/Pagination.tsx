import { QUERY_LIMIT } from '@/common/utils';

import styles from './styles.module.scss';
import { type PaginationProps } from './types';

export function Pagination({ page, dispatch, total }: PaginationProps): JSX.Element {
  return (
    <div className={styles.pagination}>
      {Boolean(page) && (
        <button
          type="button"
          aria-label="Left"
          className={styles.pag_left}
          onClick={() => {
            dispatch({ type: 'SET_PAGE', page: page - 1 });
          }}
        />
      )}
      <div className={styles.pages}>
        <span>{page + 1}</span>-
        <span>{Math.floor(total.current / QUERY_LIMIT) + (total.current % QUERY_LIMIT > 0 ? 1 : 0)}</span>
      </div>
      {page + 1 < Math.floor(total.current / QUERY_LIMIT) + (total.current % QUERY_LIMIT > 0 ? 1 : 0) && (
        <button
          type="button"
          aria-label="Right"
          className={styles.pag_right}
          onClick={() => {
            dispatch({ type: 'SET_PAGE', page: page + 1 });
          }}
        />
      )}
    </div>
  );
}
