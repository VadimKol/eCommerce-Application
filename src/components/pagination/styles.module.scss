.pagination {
  display: flex;
  justify-content: center;
  margin-top: 30px;
}

.pagination_box,
.pagination_box_before,
.pagination_box_after {
  display: flex;
  gap: 10px;
  align-items: center;
}

.pagination_box_before {
  &::before {
    content: "";
    display: block;
    width: 45px;
    height: 45px;
  }
}

.pagination_box_after {
  &::after {
    content: "";
    display: block;
    width: 45px;
    height: 45px;
  }
}

.pages {
  font-size: 1.5em;
  background-color: var(--background-color);
  border-radius: 6px;
  box-shadow: 0 0 4px 4px var(--background-color);
}

.pag_left,
.pag_right {
  width: 45px;
  height: 45px;
  background-color: var(--background-color);
  border: none;
  border-radius: 6px;
  box-shadow: 0 0 4px 4px var(--background-color);
  cursor: pointer;
  transition: transform 0.5s;


  &::after {
    content: "";
    display: block;
    width: 100%;
    height: 100%;
    background-color: var(--primary-color);
    transition: background-color 0.5s;
    mask-image: url("/src/assets/images/arrow_down.svg");
    mask-size: cover;
  }

  @media (hover: hover) {
    &:hover::after {
      background-color: var(--tertiary-color);
    }
  }
}

.pag_left {
  transform: rotate(90deg);

  @media (hover: hover) {
    &:hover {
      transform: rotate(90deg) scale(1.2);
    }
  }
}

.pag_right {
  transform: rotate(-90deg);

  @media (hover: hover) {
    &:hover {
      transform: rotate(-90deg) scale(1.2);
    }
  }
}
