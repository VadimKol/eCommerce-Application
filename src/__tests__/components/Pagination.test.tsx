import { render } from '@testing-library/react';
import React from 'react';

import { Pagination } from '@/components/pagination/Pagination';

it('renders correctly', () => {
  const someDispatch = (): void => {};
  const total: React.MutableRefObject<number> = { current: 0 };
  const useRef = jest.fn().mockImplementation(() => [total]);
  jest.spyOn(React, 'useRef').mockImplementation(useRef);

  const { container } = render(<Pagination page={0} dispatch={someDispatch} total={total} />);

  expect(container).toMatchSnapshot();
});
