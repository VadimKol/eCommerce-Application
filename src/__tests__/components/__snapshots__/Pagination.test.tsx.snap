// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="pagination"
  >
    <div
      class="pagination_box_after"
    >
      <div
        class="pages"
      >
        <span>
          1
        </span>
        -
        <span>
          0
        </span>
      </div>
    </div>
  </div>
</div>
`;
